packages/aws-cdk-lib/aws-codepipeline/test/pipeline.test.ts

```diff
@@ -496,6 +497,545 @@ describe('', () => {
+        output: sourceArtifact,
+        connectionArn: 'connection',
+        owner: 'owner',
+        repo: 'repo',
+      });
+
+      expect(() => {
+        new codepipeline.Pipeline(stack, 'Pipeline', {
+          pipelineType: codepipeline.PipelineType.V2,
+          triggers: [{
+            providerType: codepipeline.ProviderType.CODE_STAR_SOURCE_CONNECTION,
+            gitConfiguration: {
+              sourceAction,
+              pushFilter: [{
+                excludedTags: ['exclude1', 'exclude2', 'exclude3', 'exclude4', 'exclude5', 'exclude6', 'exclude7', 'exclude8', 'exclude9'],
+                includedTags: ['include1', 'include2'],
+              }],
+            },
+          }],
+        });
+      }).toThrow(/maximum length of excludedTags is 8, got 9 at triggers\[0\].gitConfiguration.pushFilter\[0\]/);
+    });
+
+    test('throw if length of excludes is greater than 8', () => {
+      const stack = new cdk.Stack();
+      const sourceArtifact = new codepipeline.Artifact();
+      const sourceAction = new CodeStarConnectionsSourceAction({
+        actionName: 'CodeStarConnectionsSourceAction',
+        output: sourceArtifact,
+        connectionArn: 'connection',
+        owner: 'owner',
+        repo: 'repo',
+      });
+
+      expect(() => {
+        new codepipeline.Pipeline(stack, 'Pipeline', {
+          pipelineType: codepipeline.PipelineType.V2,
+          triggers: [{
+            providerType: codepipeline.ProviderType.CODE_STAR_SOURCE_CONNECTION,
+            gitConfiguration: {
+              sourceAction,
+              pushFilter: [{
+                excludedTags: ['exclude1', 'exclude2'],
+                includedTags: ['include1', 'include2', 'include3', 'include4', 'include5', 'include6', 'include7', 'include8', 'include9'],
+              }],
+            },
+          }],
+        });
+      }).toThrow(/maximum length of includedTags is 8, got 9 at triggers\[0\].gitConfiguration.pushFilter\[0\]/);
+    });
+
+    test('throw if provider of sourceAction is not \'CodeStarSourceConnection\'', () => {
+      const stack = new cdk.Stack();
+      const sourceArtifact = new codepipeline.Artifact();
+      const sourceAction = new FakeSourceAction({
+        actionName: 'FakeSource',
+        output: sourceArtifact,
+      });
+
+      expect(() => {
+        new codepipeline.Pipeline(stack, 'Pipeline', {
+          pipelineType: codepipeline.PipelineType.V2,
+          triggers: [{
+            providerType: codepipeline.ProviderType.CODE_STAR_SOURCE_CONNECTION,
+            gitConfiguration: {
+              sourceAction,
+              pushFilter: [{
+                excludedTags: ['exclude1', 'exclude2'],
+                includedTags: ['include1', 'include2'],
+              }],
+            },
+          }],
+        });
+      }).toThrow(/provider for actionProperties in sourceAction must be 'CodeStarSourceConnection', got 'Fake' at triggers\[0\]/);
```

Comment:
I'm confused by this test:

1) this looks like an additional requirement, is it documented anywhere
2) it says got 'Fake'. I don't see where we specified the provider to be `fake` so this confuses me.